packages/odie-client/src/context/index.tsx

```diff
@@ -209,13 +213,21 @@ const OdieAssistantProvider: FC< OdieAssistantProviderProps > = ( {
 				// Append new messages at the end
 				return {
 					chat_id: prevChat.chat_id,
+					type: newMessages.find( ( m ) => m.role === 'agent' ) ? 'human' : prevChat.type,
 					messages: [ ...filteredMessages, ...newMessages ],
 				};
 			} );
 		},
 		[ setChat ]
 	);
 
+	useEffect( () => {
+		addMessengerListener( ( message: any ) => {
+			const translatedMessage = translateMessage( message );
```

Comment:
I'd call this `transformMessage` or `normalizeMessage`. Translate seems misleading.